### PR TITLE
[WW-5065] Defines a new flag to control appending params - cherry-pick

### PR DESCRIPTION
**Pull request by lukaszlenart**
Refs https://github.com/apache/struts/pull/405
Refs [WW-5065](https://issues.apache.org/jira/browse/WW-5065)

### CODE REVIEW

File: core/src/main/java/com/opensymphony/xwork2/config/providers/XWorkConfigurationProvider.java [hunk 1]
```diff
@@ -107,6 +107,7 @@
 import com.opensymphony.xwork2.validator.DefaultValidatorFileParser;
 import com.opensymphony.xwork2.validator.ValidatorFactory;
 import com.opensymphony.xwork2.validator.ValidatorFileParser;
+import com.sun.org.apache.xpath.internal.operations.Bool;
```
Comment (davelnewton) [inline core/src/main/java/com/opensymphony/xwork2/config/providers/XWorkConfigurationProvider.java:110]: Seems spurious

File: core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java [hunk 2]
```diff
@@ -192,7 +217,7 @@ protected String convertParam(String val, Map<String, String> vars) {
             c = val.charAt(x);
             if (x < len - 2 && 
                     c == '{' && '}' == val.charAt(x+2)) {
-                varVal = (String)vars.get(String.valueOf(val.charAt(x + 1)));
+                varVal = vars.get(String.valueOf(val.charAt(x + 1)));
```
Comment (davelnewton) [inline core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java:220]: (NON-BLOCKER) Never thought about it or ran in to it, but do we throw a startup message or error if it's something *other* than 0-9 or has a spurious `{42}` or `{42` or is that just considered part of a valid, if silly, path?

File: core/src/main/java/com/opensymphony/xwork2/config/providers/XWorkConfigurationProvider.java (hunk 1 above)
Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/config/providers/XWorkConfigurationProvider.java]: 👍

File: core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java (hunk 2 above)
Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java:220]: Hm... good point, I think we can extract `OgnlTextParser` logic and reuse it here

Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java:220]: https://issues.apache.org/jira/browse/WW-5073

**Pull request merged**